app/code/Magento/MediaGalleryUi/Test/Mftf/Test/AdminMediaGalleryUploadCategoryImageTest.xml

```diff
@@ -40,6 +37,7 @@
         <actionGroup ref="AdminSaveCategoryFormActionGroup" stepKey="saveCategoryForm"/>
         <actionGroup ref="AdminOpenMediaGalleryFromCategoryImageUploaderActionGroup" stepKey="openMediaGalleryFromImageUploader"/>
         <actionGroup ref="ResetAdminDataGridToDefaultViewActionGroup" stepKey="resetAdminDataGridToDefaultView"/>
+        <waitForLoadingMaskToDisappear stepKey="waitForLoadingMaskToDisappear"/>
         <actionGroup ref="AdminMediaGalleryAssertImageInGridActionGroup" stepKey="assertImageInGrid">
```

Comment:
Just a minor recommendation. If you take a look at `AdminEnhancedMediaGalleryImageActionsSection.imageInGrid` element selector, you will find that it might be too complex and causes fluky tests in the result. If we could make the selector simpler somehow (or introduce a new one). Like asserting the title of element with `class="preview-image"` or so. This change may stabilize such tests as this one.

Hope it will help.